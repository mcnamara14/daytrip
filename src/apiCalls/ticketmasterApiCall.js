import { ticketmasterApiKey } from './apiKeys/ticketmasterApiKey';

export const ticketmasterApiCallRecentEvents = async (city, state, timeNow, timeIn2Days) => {
  const url = `https://app.ticketmaster.com/discovery/v2/events.json?startDateTime=${timeNow}&endDateTime=${timeIn2Days}&sort=date,name,asc&city=${city}&stateCode=${state}&apikey=${ticketmasterApiKey}`;

  try {
    const response = await fetch(url);
    const data = await response.json();
    const events = data._embedded.events;

    return events
  } catch (error) {
      alert(error)
  }
}





