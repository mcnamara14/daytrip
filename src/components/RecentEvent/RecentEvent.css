.recentEvents {
  width: 100%;
  height: 100%;
  box-shadow: inset 0px 9px 8px -6px #eee;
}

.upcomingEvents {
  width: 86%;
  margin: 0 auto;
  text-align: left;
  padding-top: 50px;
}

.upcomingEvents h2 {
  margin: 0 0 50px 0;
  color: #494949;
  text-transform: uppercase;
  font-size: 1.4rem;
}

.upcomingEventsContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.recentEvent {
  text-align: center;
  width: 24%;
  height: 520px;
  text-align: left;
  margin-bottom: 30px;
  border-radius: 5px;
}

.recentEvent h3 {
  color: #7c7c7c;
  text-transform: uppercase;
  font-size: 1rem;
  display: inline;
}

.recentTitle {
  display: flex;
  height: 40px;
  align-items: flex-end;
}

.eventImage {
  width: 90%;
  height: 190px;
  border-radius: 5px 5px 0 0;
  display: block;
  border-radius: 5px;
  background-size: cover;
  background-position: center;
  border: 1px solid #eee;
}

.recentEvent .price {
  color: #4195ce;
  font-weight: 500;
  font-size: 1.2rem;
}

.recentEvent .date {
  color: #999;
  font-size: 0.8rem;
  margin-bottom: 20px;
  margin-top: 0px;
}

.recentEvent .venue {
  color: #444;
  font-weight: 500;
  font-family: "Exo 2";
  font-size: 0.9rem;
  margin-bottom: 10px;
}

.recentEvent button {
  background: #ccc;
  padding: 10px 25px;
  font-size: 0.9rem;
  color: #fff;
  border: none;
  border-radius: 3px;
  margin-top: 35px;

}

hr {
  width: 53px;
  margin-left: 0;
  border-bottom: 0;
  border-top: 1px solid #ccc;
}

.eventInfo {
  padding: 15px 0;
}