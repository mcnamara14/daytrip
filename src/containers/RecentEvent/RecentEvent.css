.recentEvents {
  width: 100%;
  height: 100%;
  box-shadow: inset 0px 9px 8px -6px #eee;
}

.upcomingEvents {
  width: 86%;
  margin: 0 auto;
  text-align: left;
  padding-top: 50px;
}

.upcomingEvents h2 {
  margin: 0 0 50px 0;
  color: #e2604b;
  font-size: 1.4rem;
  text-transform: none;
}

.upcomingEventsContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.recentEvent {
  text-align: center;
  width: 23%;
  height: 520px;
  text-align: left;
  margin-bottom: 30px;
  box-shadow: 1px 1px 1px #eee;
  position: relative;
}

.recentEvent img {
  height: 18px;
  margin-right: 10px;
}

.recentEvent h3 {
  color: #555;
  text-transform: uppercase;
  font-size: 1rem;
  display: inline;
  font-weight: 600;
}

.recentTitle {
  display: flex;
  margin-bottom: 20px;
  height: 60px;
  text-overflow: ellipsis;
  overflow: hidden;
}

.eventImage {
  width: 86%;
  margin: 20px auto 0 auto;
  height: 190px;
  box-shadow: 0 1px 1px #eee;
  display: block;
  background-size: cover;
  background-position: center;
  border: 1px solid #eee;
}


.recentEvent .date {
  color: #b8b8b8;
  font-size: 1rem;
  margin-bottom: 20px;
  margin-top: 0px;
}

.recentEvent .venue {
  color: #888;
  margin-bottom: 0px;
  font-size: 0.9rem;
  height: 40px;
  font-family: helvetica;
  font-weight: bold;
  margin-top: 0px;
  border-radius: 18px;
  border-radius: 3px;
  display: flex;
  align-items: center;
}

.recentEvent button {
  background: rgba(116,87,117,1);
  padding: 13px 35px;
  font-size: 0.9rem;
  color: #fff;
  margin-top: 30px;
  margin-left: 20px;
  border: none;
  box-shadow: 1px 1px 1px #ccc;
  border-radius: 3px;
  position: absolute;
  bottom: 15px;
}

hr {
  width: 53px;
  margin-left: 0;
  border-bottom: 0;
  margin-bottom: 25px;
  border-top: 1px solid #ccc;
}

.eventInfo {
  padding: 20px;
  font-size: 0.9rem;
  color: #b8b8b8;
}

.recentEventVenue {
  margin-bottom: 10px;
  margin-top: 30px;
  display: flex;
}

,.recentEventPrice {
  display: flex;
}

.recentEventDate {
  color: #888;
  font-size: 0.95rem;
  display: flex;
}
