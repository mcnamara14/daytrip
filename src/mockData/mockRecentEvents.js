export const mockDirtyRecentEvents = {
  events: [
    {
      title: "T Swift",
      image: "https://s1.ticketm.allswifty.jpg",
      price: "$300",
      venue: "Boulder Theater",
      date: "2018-05-30 8:00 PM"
    }
  ]
};

