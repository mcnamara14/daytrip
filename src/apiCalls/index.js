import { ticketmasterApiCall } from './ticketmasterApiCall'; 
import { yelpApiCall } from './yelpApiCall'; 

export {
  ticketmasterApiCall,
  yelpApiCall
};