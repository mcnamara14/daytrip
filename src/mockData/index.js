import { mockDirtyRecentEvents } from './mockRecentEvents';
import { mockCleanRestaurantAndBarData } from './mockRestaurantAndBarData';

export {
  mockDirtyRecentEvents,
  mockCleanRestaurantAndBarData
}