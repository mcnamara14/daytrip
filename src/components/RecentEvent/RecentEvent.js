import React, { Component } from 'react';
import { connect } from 'react-redux';
import './RecentEvent.css';
import { storeSelectedEvent } from '../../actions/index';
import { ticketmasterFetchSelectedEvent } from '../../apiCalls/index';

class RecentEvent extends Component {
  handleRecentClick = async (eventId) => {
    const selectedEvent = await ticketmasterFetchSelectedEvent(eventId);
    const event = selectedEvent[0]

    this.props.storeSelectedEvent(event);
  }

  render() {
    const { image, title, venue, date, price, id } = this.props;
    const backgroundImage = {backgroundImage: "url(" + image + ")"};

    return (
      <div className="recentEvent">
        <div className="eventInfo">
          <div className="recentTitle">
            <h3>{title}</h3>
          </div>
          <p className="venue">{venue}</p>
          <p className="date">{date}</p>
          <hr/>
          <p className="price">{price}</p>
        </div>
        <div className="eventImage" style={ backgroundImage } ></div>
        <button onClick={() => this.handleRecentClick(id)}>Select</button>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  storeSelectedEvent: (event) => dispatch(storeSelectedEvent(event))
})

export default connect(null, mapDispatchToProps)(RecentEvent);