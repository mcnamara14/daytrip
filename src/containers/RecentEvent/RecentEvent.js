import React, { Component } from 'react';
import { connect } from 'react-redux';
import './RecentEvent.css';
import PropTypes from 'prop-types';
import { storeSelectedEvent } from '../../actions';
import { fetchSelectedEvent } from '../../apiCalls';

export class RecentEvent extends Component {
  handleRecentClick = async (eventId) => {
    const selectedEvent = await fetchSelectedEvent(eventId);
    const event = selectedEvent[0];

    this.props.storeSelectedEvent(event);
  }

  render() {
    const { image, title, venue, date, price, id } = this.props;
    const backgroundImage = {backgroundImage: "url(" + image + ")"};

    return (
      <div className="recentEvent">
        <div className="eventImage" style={ backgroundImage } ></div>
        <div className="eventInfo">
          <div className="recentTitle">
            <h3>{title}</h3>
          </div>
          <p className="price">{price}</p>
          <hr/>
          <p className="venue">{venue}</p>
          <p className="date">{date}</p>
        </div>
        <button onClick={() => this.handleRecentClick(id)}>Select</button>
      </div>
    );
  }
}

RecentEvent.propTypes = {
  storeSelectedEvent: PropTypes.func,
  image: PropTypes.string,
  title: PropTypes.string,
  venue: PropTypes.string,
  date: PropTypes.string,
  price: PropTypes.string,
  id: PropTypes.string
};

export const mapDispatchToProps = (dispatch) => ({
  storeSelectedEvent: (event) => dispatch(storeSelectedEvent(event))
});

export default connect(null, mapDispatchToProps)(RecentEvent);