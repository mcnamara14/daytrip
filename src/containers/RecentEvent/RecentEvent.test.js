import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { RecentEvent, mapDispatchToProps } from './RecentEvent';
import { fetchSelectedEvent } from '../../apiCalls';
jest.mock('../../apiCalls');
import { mockDirtyRecentEvents } from '../../mockData/mockRecentEvents'

describe('RecentEvent', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<RecentEvent />);
  
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleRecentClick', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<RecentEvent storeSelectedEvent={jest.fn()} />);
      fetchSelectedEvent.mockImplementation(() => [{title: 'T-Swift'}])
    })

    it('should call fetchSelectedEvent with the correct argument', async () => {
      const expected = '12345';

      await wrapper.instance().handleRecentClick(expected);

      expect(fetchSelectedEvent).toHaveBeenCalledWith(expected);
    })

    it('should call storeSelectedEvent with the correct argument', async () => {
      const expected = {title: 'T-Swift'};

      await wrapper.instance().handleRecentClick(expected);

      expect(wrapper.instance().props.storeSelectedEvent).toHaveBeenCalledWith(expected);
    })
  });

  describe('Event button click', () => {
    it('should call handleRecentClick with correct argument when button is clicked to add event', () => {
      const wrapper = shallow(<RecentEvent id={'12345'} storeSelectedEvent={jest.fn()} />);

      wrapper.instance().handleRecentClick = jest.fn();

      wrapper.find('button').simulate('click');
      expect(wrapper.instance().handleRecentClick).toHaveBeenCalledWith('12345');
    })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with the correct params on storeSelectedEvent', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const mockEvent = mockDirtyRecentEvents.events[0];
      const mockAction = {
        type: 'STORE_SELECTED_EVENT',
        event: mockEvent
      };

      mappedProps.storeSelectedEvent(mockEvent);

      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  })
});
