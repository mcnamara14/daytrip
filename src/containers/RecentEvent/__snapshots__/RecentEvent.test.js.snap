// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="eventImage"
          style={
                    Object {
                              "backgroundImage": "url(undefined)",
                            }
          }
/>,
        <div
          className="eventInfo"
>
          <h3 />
          <p
                    className="price"
          />
          <hr />
          <p
                    className="venue"
          />
          <p
                    className="date"
          />
          <button>
                    Select
          </button>
</div>,
      ],
      "className": "recentEvent",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "eventImage",
          "style": Object {
            "backgroundImage": "url(undefined)",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3 />,
            <p
              className="price"
/>,
            <hr />,
            <p
              className="venue"
/>,
            <p
              className="date"
/>,
            <button>
              Select
</button>,
          ],
          "className": "eventInfo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "price",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "venue",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "date",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Select",
            },
            "ref": null,
            "rendered": "Select",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="eventImage"
            style={
                        Object {
                                    "backgroundImage": "url(undefined)",
                                  }
            }
/>,
          <div
            className="eventInfo"
>
            <h3 />
            <p
                        className="price"
            />
            <hr />
            <p
                        className="venue"
            />
            <p
                        className="date"
            />
            <button>
                        Select
            </button>
</div>,
        ],
        "className": "recentEvent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "eventImage",
            "style": Object {
              "backgroundImage": "url(undefined)",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3 />,
              <p
                className="price"
/>,
              <hr />,
              <p
                className="venue"
/>,
              <p
                className="date"
/>,
              <button>
                Select
</button>,
            ],
            "className": "eventInfo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "price",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "venue",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "date",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Select",
              },
              "ref": null,
              "rendered": "Select",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
